Tests/AK/TestBuiltinWrappers.cpp

```diff
@@ -65,3 +67,40 @@ TEST_CASE(wrapped_count_required_bits)
     EXPECT_EQ(count_required_bits(0b1111u), 4ul);
     EXPECT_EQ(count_required_bits(NumericLimits<u32>::max()), sizeof(u32) * 8);
 }
+
+RANDOMIZED_TEST_CASE(count_leading_zeroes)
+{
+    //    count_leading_zeroes(0b000...0001000...000)
+    // == count_leading_zeroes(0b000...0001___...___) (where _ is 0 or 1)
+    GEN(e, Gen::unsigned_int(0, 63));
+    auto power_of_two = 1ULL << e;
+
+    // We add random one-bits below the leftmost (and only) one-bit.
+    // This shouldn't change the output of count_leading_zeroes because
+    // the function should only care about the most significant one.
+    GEN(below, Gen::unsigned_int(0, power_of_two - 1));
+    auto n = power_of_two + below;
+
+    EXPECT_EQ(count_leading_zeroes(n), count_leading_zeroes(power_of_two));
+}
+
+RANDOMIZED_TEST_CASE(count_required_bits)
+{
+    // count_required_bits(n) == log2(n) + 1
+
+    // log2(0) is -infinity, we don't care about that
+    GEN(n, Gen::unsigned_int());
+    ASSUME(n > 0);
+
+    // max due to edge case for n=0
+    size_t expected = max(1, AK::log2(static_cast<double>(n)) + 1);
+
+    EXPECT_EQ(count_required_bits(n), expected);
+}
+
+RANDOMIZED_TEST_CASE(bit_scan_forward_count_trailing_zeroes)
+{
+    // behaviour for 0 differs so we skip it
```

Comment:
```suggestion
    // Behaviour for 0 differs, so we skip it.
```

Not sure about the comma, but comments should at the very least start with a capital letter and end with a period if they are meant to resemble a sentence (excluding variable names, or quotes from elsewhere).